netplugin/agent/state_event.go

```diff
@@ -250,6 +252,39 @@ func processEpState(netPlugin *plugin.NetPlugin, opts core.InstanceInfo, epID st
 	return err
 }
 
+// processRemoteEpState updates endpoint state
+func processRemoteEpState(netPlugin *plugin.NetPlugin, opts core.InstanceInfo, epCfg *mastercfg.CfgEndpointState, isDelete bool) error {
+	// take a lock to ensure we are programming one event at a time.
+	netPlugin.Lock()
+	defer func() { netPlugin.Unlock() }()
+
+	if !checkRemoteHost(epCfg.VtepIP, epCfg.HomingHost, opts.HostLabel) {
+		// Skip local endpoint update, as they are handled directly in dockplugin
+		return nil
+	}
+
+	var err error
```

Comment:
this var can be completely removed if you switch the err checks to be inline in the `if` statements